rpm/0084-sailfishos-gecko-dev-Disallow-page-zooming-if-the-me.patch

```diff
@@ -0,0 +1,43 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Andrew den Exter <andrew.den.exter@jolla.com>
+Date: Thu, 11 Nov 2021 06:51:59 +0000
+Subject: [PATCH] [sailfishos][gecko-dev] Disallow page zooming if the meta
+ viewport scale is locked to 1. JB#56149 OMP#JOLLA-499
+
+---
+ layout/base/nsLayoutUtils.cpp | 13 ++++++++++++-
+ 1 file changed, 12 insertions(+), 1 deletion(-)
+
+diff --git a/layout/base/nsLayoutUtils.cpp b/layout/base/nsLayoutUtils.cpp
+index 2cd2903a8fc8..59981a1ca9c7 100644
+--- a/layout/base/nsLayoutUtils.cpp
++++ b/layout/base/nsLayoutUtils.cpp
+@@ -694,13 +694,24 @@ bool nsLayoutUtils::AsyncPanZoomEnabled(nsIFrame* aFrame) {
+   return widget->AsyncPanZoomEnabled();
+ }
+ 
++
++static bool DocumentHasFixedUnityZoom(const mozilla::dom::Document* aDocument)
++{
++  if (!aDocument) {
++    return false;
++  }
++  nsresult rv;
++  ViewportMetaData metaData = aDocument->GetViewportMetaData();
++  return metaData.mMinimumScale.ToDouble(&rv) == 1.0 && metaData.mMaximumScale.ToDouble(&rv) == 1.0;
```

Comment:
It's enough that min and max are equal to fix zooming. Further, following conditions should disable zooming as well:

```
metaData.mUserScalable.EqualsLiteral("0") or
metaData.mUserScalable.EqualsLiteral("no") or
metaData.mUserScalable.EqualsLiteral("false")
```

Further, there places like (e.g in ZoomContraintsClient.cpp):
```
  constraints.mAllowZoom = aViewportInfo.IsZoomAllowed() &&
                           nsLayoutUtils::AllowZoomingForDocument(aDocument);
```
Hence, thinking that maybe worth adjusting Document::GetViewportInfo as well so that effectiveZoomFlag ends up being nsViewportInfo::ZoomFlag::DisallowZoom also when min & max are equal (Document.cpp ~ 9597).

Reply:
Document::GetViewportInfo returns a ViewportMetaData. nsViewport is returned by Document::GetViewportInfo which requires display dimensions as arguments.

Reply:
You missed my point. I meant that effectiveZoomFlag inside Docoment::GetViewportInfo currently does not end up being nsViewportInfo::ZoomFlag::DisallowZoom (Document.cpp:9635) when minimum and maximum are the same (mAllowZoom = false => Document.cpp:9601).

I think this is just in correct behaviour.